synapse/storage/databases/main/roommember.py

```diff
@@ -350,6 +350,51 @@ def _get_rooms_for_local_user_where_membership_is_txn(
 
         return results
 
+    @cached()
+    async def get_local_current_membership_for_user_in_room(
+        self, user_id: str, room_id: str
+    ) -> Tuple[Optional[str], Optional[str]]:
+        """Retrieve the current local membership state and event ID for a user in a room.
+
+        Args:
+            user_id: The ID of the user.
+            room_id: The ID of the room.
+
+        Returns:
+            A tuple of (membership_type, event_id). Both will be None if a
+                room_id/user_id pair is not found.
+
+        """
+        return await self.db_pool.runInteraction(
+            "get_local_current_membership_for_user_in_room",
+            self._get_local_current_membership_for_user_in_room_txn,
+            user_id,
+            room_id,
+        )
```

Comment:
I think this can be replaced with:

```python
simple_select_one(
    "local_current_membership",
    {"room_id": room_id, "user_id": user_id},
    ("membership", "event_id"),
    allow_none=True,
    desc="get_local_current_membership_for_user_in_room",
)
```

Reply:
Ah yes, good point! I've done this, and moved the paranoia check into the public function as well.

I'm not 100% sure if that paranoia check is necessary, as a remote user would just cause the query to return `None`. That does mean the user will get "Not a known room" though, which isn't as useful an error :shrug: